### PR TITLE
test: increase coverage for metadata database module

### PR DESCRIPTION
**Pull request by Riddhi8077**
This PR increases test coverage for the metadata database module.

Functions Covered:
- db_create_metadata_table
- db_get_metadata
- db_update_metadata

Changes:
- Added deterministic unit tests for table creation and default row insertion.
- Covered metadata retrieval and update flows.
- Tested update behavior using both internal and external database cursors.
- Included JSON decoding edge case validation.
- No production logic was modified.

Coverage:
Before: 14%
After: 88%

How to run tests:
pytest --cov=app --cov-report=term-missing

## Summary by CodeRabbit

* **Tests**
  * Added isolated database tests covering connection behavior, PRAGMA settings, commit/rollback semantics, connection lifecycle, and metadata lifecycle (create/read/update and handling corrupted data). Replaced the previous global shared test setup with a temporary-database fixture that scopes setup/teardown to a temp file.

* **Chores**
  * Minor formatting adjustment to the repository ignore configuration.

### CODE REVIEW

File: backend/tests/database/test_connection.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+def test_connection_closed_after_context(temp_db_path):
+    """
+    Ensure connection is closed after exiting context manager.
+    """
+    with get_db_connection() as conn:
+        pass
+
+    with pytest.raises(sqlite3.ProgrammingError):
+        conn.execute("SELECT 1;")
```
[the remark this answers is not in the document]
Comment (RiddhiPachehara) [reply backend/tests/database/test_connection.py:74]: Thank you for the review.
Removed duplicate .gitignore entries and verified test structure follows pytest conventions (fixtures in conftest.py, tests in dedicated modules).
Please let me know if any further refinements are needed.

[the remark this answers is not in the document]
Comment (RiddhiPachehara) [reply backend/tests/database/test_connection.py:74]: Thank you for the detailed review.

I have re-verified and cleaned up the structure to ensure there is no duplication. The shared setup logic remains in conftest.py, and the connection tests are defined only in their dedicated test module. The fixture structure now follows standard pytest conventions, and tests run without duplication.

If you are still seeing any duplication or structural issue on your side, please feel free to point to the specific file/line or suggest the exact change. I’m also happy for you to propose or plan the adjustments you think would best align with the project standards.

Looking forward to your feedback.

[the remark this answers is not in the document]
Comment (RiddhiPachehara) [reply backend/tests/database/test_connection.py:74]: Thank you again for the detailed review.
Removed the duplicate connection tests and fixture from conftest.py.
The file now contains only the session-scoped setup fixture, and the connection tests remain exclusively in test_connection.py following pytest conventions.
Please let me know if anything else needs refinement.

[the remark this answers is not in the document]
Comment (RiddhiPachehara) [reply backend/tests/database/test_connection.py:74]: Yes I Understood, Now all tests pass (109/109).
Refactored DB test setup to ensure proper isolation and fixture ordering.

**Comment by RiddhiPachehara** [comment]
@CodeRabbitai Review

File: backend/tests/conftest.py [hunk 2]
```diff
@@ -1,44 +1,93 @@
...
-    yield  # This is where the tests run
-
-    # Teardown code runs after all tests
-    print("\n=== Running cleanup after all tests ===")
-
-    # Cleanup code here
-    if "TEST_MODE" in os.environ:
-        del os.environ["TEST_MODE"]
+    with pytest.raises(sqlite3.ProgrammingError):
+        conn.execute("SELECT 1;")
```
[the remark this answers is not in the document]
Comment (RiddhiPachehara) [reply backend/tests/conftest.py]: @CodeRabbitai Resume

**Comment by SIDDHANTCOOKIE** [comment]
Closing this as unreviewed issue
